py_hamt/hamt.py

```
@@ -39,568 +38,618 @@ def extract_bits(hash_bytes: bytes, depth: int, nbits: int) -> int:
     def serialize(self) -> bytes:
         return dag_cbor.encode(self.data)
 
     @classmethod
     def deserialize(cls, data: bytes) -> "Node":
         try:
-            decoded = dag_cbor.decode(data)
+            decoded_data = dag_cbor.decode(data)
         except:  # noqa: E722
             raise Exception(
                 "Invalid dag-cbor encoded data from the store was attempted to be decoded"
             )
```

Ruff (0.11.9)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)